File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    let process = priorities.map((pro,idx)=>{return{pro,idx}});
    let run=0;
    while(true){
        let flag = false;
        const target = process.splice(0,1)[0];
        
        for(let i = 0;i<process.length;i++){
            if(process[i].pro > target.pro){
                process.push(target);
                flag=true;
                break;
            }
        }
        if(!flag){
            run++;
            console.log(target,run);
            if(target.idx === location) {
                return run;
            }
            
        }
        if(process.length === 0) break;
    }
}
// 프로세스를 선택(배열의 첫번째)
// 선택한 프로세스의 우선순위를 다른 프로세스의 우선순위 들과 비교
// 선택한 프로세스의 우선순위보다 큰 우선 순위가 있다면 해당 프로세스를 배열 맨뒤에 추가
// 

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <vector>
#include <queue>
#include <utility>
using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    queue<pair<int,int>> setPriorities;
    for(int i =0;i<priorities.size();i++){
        setPriorities.push(make_pair(priorities[i],i));
    }
    while(1){
        pair<int,int> p = setPriorities.front(); // 맨앞에 값 가져오기
        bool flag = true;
        
        queue<pair<int,int>> tempPriorities=setPriorities; // 큐를 복사
        
        while(!tempPriorities.empty()){// 복사한 큐를 돌면서 더 큰 값이 있나 찾기
            pair<int,int> pp = tempPriorities.front();
            
            if(p.first < pp.first) {
                setPriorities.push(p);
                flag = false;
                break;
            }
            else {
                tempPriorities.pop();
            }
        }
        setPriorities.pop();
        if(flag) {
            answer++; //값을 찾았으므로 answer
             if(p.second == location) break;
        }
        
        
       
    }
    return answer;
}
// 한개씩 가져와서 